website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,244 @@
+        chd_dict = {
+            'origin_product_id': curr_product_id.id,
+            'partner_id': request.env.user.partner_id.id,
+            'state': 'config',
+            'quantity': form_data['quantity']
+            }
+        # dynamic and fixed size types
+        if form_data['product_id_chd_size_type'] == "fixed":
+            chd_dict['size_id'] = form_data['size']
+            chd_size = request.env['chd.size'].browse([
+                int(form_data['size'])
+                ])
+            chd_dict['width'] = chd_size.width
+            chd_dict['height'] = chd_size.height
+        else:
+            chd_dict['width'] = form_data['width']
+            chd_dict['height'] = form_data['height']
+        # if there aren't any errors, upload the image
+        message = ''
+        has_image = product_template_model.browse([
+            int(form_data['product_id'])
+            ]).chd_configurator_has_image
+        if has_image:
+            try:
+                import base64
+                fileitem = form_data['pic']
+                # add uploaded image to configurator
+                chd_dict['image'] = base64.b64encode(fileitem.stream.read())
+                chd_dict['image_filename'] = fileitem.filename
+                message = 'image uploaded successfully'
+            except:
+                message = 'Problems uploading your image, please contact us'
+        if message == '':
+            message = 'no image needed for this product'
+        for key in form_data:
+            if ('pricecomponent_' in key):
+                # if it is of type "string" the value is the index of the
+                # selection in pricecomponent_name, the id encoded in the key
+                # if it is of type numerical the value is the actual value of
+                # the pricecomponent field, the id is encoded in the key
+                if ('pricecomponent_string' in key) or ('pricecomponent_int' in key):
+                    pricecomponent_id = int(key.split('_')[3])
+                    pricecomponent_value = form_data[key]
+                # the configurator stores the attributes in a
+                # dictionary in the "attributes" field ,
+                all_attributes[
+                    '_attribute_' + str(pricecomponent_id)
+                    ] = int(pricecomponent_value)
+
+        # add attributes to the configurator
+        chd_dict['attributes'] = str(all_attributes)
+        # dictionary is complete, create configurator
+        try:
+            new_chd = request.env['chd.product_configurator'].create(chd_dict)
+        except orm.except_orm as e:
+            errormsg = e.value
+        except AssertionError as a:
+            errormsg = a.value
+        # add accessories and price components selections to the configurator
+        list = []
+        for key in form_data:
+            # get only accessories that have been checked
+            if ('accessoryid_' in key) and form_data[key] != 0:
```

Comment:
you intify `form_data[key]`, so you probably want to check for `int(form_data[key]) != 0` here? But shouldn't it be `int(form_data[key]) > 0` anyway?